src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenStringConcat.cs

```diff
@@ -358,18 +358,84 @@ static void Main()
             comp.VerifyDiagnostics();
             comp.VerifyIL("Test.Main", @"
 {
-  // Code size       35 (0x23)
+  // Code size       44 (0x2c)
```

Comment:
how does this compare to the IL produced for the pre-Roslyn compiler?

Reply:
In this specific case, pre-Roslyn generates the same as Roslyn did before this PR: a straight call to `string.Concat(object)` with no subsequent null-check for an object, and `S ?? ""` for a string.

Pre-Roslyn generates the same for `O + null + null` and `S + null + null` (whereas Roslyn generated `string.Concat(O) ?? ""` for the first case).